solr/solrj/src/java/org/apache/solr/client/solrj/impl/LBHttp2SolrClient.java

```diff
@@ -88,23 +89,63 @@ public LBHttp2SolrClient(Http2SolrClient httpClient, String... baseSolrUrls) {
     this.httpClient = httpClient;
   }
 
+  public LBHttp2SolrClient(Http2SolrClient httpClient, List<String> baseSolrUrls, Builder builder) {
+    super(baseSolrUrls);
+    this.httpClient = httpClient;
+    if (builder.responseParser != null) {
+      this.parser = builder.responseParser;
+    }
+    if (builder.requestWriter != null) {
+      this.httpClient.setRequestWriter(builder.requestWriter);
+    }
+  }
+
   @Override
   protected SolrClient getClient(String baseUrl) {
     return httpClient;
   }
 
+  /**
+   * Note: This setter method is <b>not thread-safe</b>.
+   *
+   * @param parser Default Response Parser chosen to parse the response if the parser were not
+   *     specified as part of the request.
+   * @see org.apache.solr.client.solrj.SolrRequest#getResponseParser()
+   * @deprecated use {@link LBHttp2SolrClient.Builder#withResponseParser(ResponseParser)} instead
+   */
+  @Deprecated
   @Override
   public void setParser(ResponseParser parser) {
     super.setParser(parser);
     this.httpClient.setParser(parser);
   }
 
+  @Override
+  public ResponseParser getParser() {
+    return httpClient.getParser();
```

Comment:
this `getParser()` method overrides the one in the super class, then there is no need for the `super.setParser(parser);` on line 119 above. I would actually mark the LBSolrClient#setParser() method as deprecated too, to convey a clearer message.

Reply:
Okay, so, just to be clear, I'm going to remove those two methods on the LBSolrClient...    In this PR ;-).     I like removing code!!!!